packages/core/src/modules/dids/methods/sov/SovDidRegistrar.ts

```diff
@@ -0,0 +1,189 @@
+
+      if (!submitterDid.startsWith('did:sov:')) {
+        throw new Error('Submitter did must be a valid did:sov did')
+      }
+
+      const unqualifiedSubmitterDid = submitterDid.replace('did:sov:', '')
+
+      // TODO: it should be possible to pass the pool used for writing to the indy ledger service.
+      // The easiest way to do this would be to make the submitterDid a fully qualified did, including the indy namespace.
+      await this.indyLedgerService.registerPublicDid(
+        agentContext,
+        unqualifiedSubmitterDid,
+        unqualifiedIndyDid,
+        verkey,
+        alias,
+        role
+      )
+
+      // Create did document
+      const didDocumentBuilder = sovDidDocumentFromDid(qualifiedSovDid, verkey)
+
+      // Add services if endpoints object was passed.
+      if (options.options.endpoints) {
+        await this.indyLedgerService.setEndpointsForDid(agentContext, unqualifiedIndyDid, options.options.endpoints)
+        addServicesFromEndpointsAttrib(
+          didDocumentBuilder,
+          qualifiedSovDid,
+          options.options.endpoints,
+          `${qualifiedSovDid}#key-agreement-1`
+        )
+      }
+
+      // Build did document.
+      const didDocument = didDocumentBuilder.build()
+
+      // FIXME: we need to update this to the `indyNamespace` once https://github.com/hyperledger/aries-framework-javascript/issues/944 has been resolved
+      const indyNamespace = this.indyPoolService.ledgerWritePool.config.id
+      const qualifiedIndyDid = `did:indy:${indyNamespace}:${unqualifiedIndyDid}`
+
+      // Save the did so we know we created it and can issue with it
+      const didRecord = new DidRecord({
+        id: qualifiedSovDid,
+        role: DidDocumentRole.Created,
+        tags: {
+          recipientKeyFingerprints: didDocument.recipientKeys.map((key) => key.fingerprint),
+          qualifiedIndyDid,
+        },
+      })
+      await this.didRepository.save(agentContext, didRecord)
+
+      return {
+        didDocumentMetadata: {
+          qualifiedIndyDid,
+        },
+        didRegistrationMetadata: {
+          indyNamespace,
+        },
+        didState: {
+          state: 'finished',
+          did: qualifiedSovDid,
+          didDocument,
+          secret: {
+            // FIXME: the uni-registrar creates the seed in the registrar method
+            // if it doesn't exist so the seed can always be returned. Currently
+            // we can only return it if the seed was passed in by the user. Once
+            // we have a secure method for generating seeds we should use the same
+            // approach
+            seed: options.secret?.seed,
+          },
+        },
+      }
+    } catch (error) {
+      return {
+        didDocumentMetadata: {},
+        didRegistrationMetadata: {},
+        didState: {
+          state: 'failed',
+          reason: `unknownError: ${error.message}`,
+        },
+      }
+    }
+  }
+
+  public async update(): Promise<DidUpdateResult> {
```

Comment:
It might be nice for these methods that are not supported to add some TSDoc to this so that you can see it before you even call the method. (maybe a simple comment like this)

```ts
/**
 * 
 * @remark did:sov cannot be updated
 * 
 */
```

Reply:
I can add these comments, but they're never called directly always throug the did registrar, which selects it based on the params you provide. So you'll never actually see this comment except for in the `SovDidRegistrar` file itself

Reply:
Okay it might be a bit too verbose then. You dont have to add them